utils/API.js

```diff
@@ -38,8 +38,17 @@ export async function	fetchCryptoPrice(from, to = 'usd') {
 	return null;
 }
 
-export async function	fetchBlockTimestamp(timestamp) {
-	const	result = await performGet(`https://api.etherscan.io/api?module=block&action=getblocknobytime&timestamp=${timestamp}&closest=before&apikey=JXRIIVMTAN887F9D7NCTVQ7NMGNT1A4KA3`);
+export async function	fetchBlockTimestamp(timestamp, network = 1) {
+	if (network === 250) {
+		const	result = await performGet(`https://api.ftmscan.com/api?module=block&action=getblocknobytime&timestamp=${timestamp}&closest=before&apikey=${process.env.FTMSCAN_API}`);
+
+		if (result) {
+			return result.result;
+		}
+		return null;
+	}
+
+	const	result = await performGet(`https://api.etherscan.io/api?module=block&action=getblocknobytime&timestamp=${timestamp}&closest=before&apikey=${process.env.ETHERSCAN_API}`);
```

Comment:
what happens when we hit etherscan API limit? It's goign to throw an error, believe me

Reply:
When we will use the AP(e)I, this will cache some of the result